### PR TITLE
Add ResourceRequirments to WorkflowController

### PR DESCRIPTION
**Pull request by gmfrasca**
- Introduces ResourceRequirements for WorkflowController in DSPA CRD
- Adds dynamic request and limit defaults for WC deployment

## The issue resolved by this Pull Request:
Resolves [RHOAIENG-8242](https://issues.redhat.com/browse/RHOAIENG-8242)

## Testing instructions
- Deploy DSPO
- Create DSPA, preferably with custom values set for `spec.workflowController.resources.requests.memory`, `spec.workflowController.resources.limits.memory`, `spec.workflowController.resources.requests.cpu`, and `spec.workflowController.resources.limits.cpu`
- Go to OpenShift Console, go to DSPA project, and click Pods
- Find pod with name starting with `ds-pipeline-workflow-controller-` and click it
- Go to Metrics tab, ensure the Memory Usage and CPU Usage graphs show correct limits
- Run several simultaneous pipelines, ensure the limits are not encroached upon

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-667`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/667/head
git checkout -b pullrequest 00ae75c614be8531149f3fa438f5707d7e0a5aa5
oc new-project opendatahub
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-667"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

File: api/v1alpha1/dspipeline_types.go [hunk 1]
```diff
@@ -331,6 +331,8 @@ type WorkflowController struct {
 	Image         string `json:"image,omitempty"`
 	ArgoExecImage string `json:"argoExecImage,omitempty"`
 	CustomConfig  string `json:"customConfig,omitempty"`
+	// Specify custom Pod resource requirements for this component.
+	Resources *ResourceRequirements `json:"resources,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:335]: Can you also update: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/samples/v2/dspa-all-fields/dspa_all_fields.yaml#L11 

looks like we never added the workflowcontroller section, could you add that and also add the other fields?

Comment (gmfrasca) [reply api/v1alpha1/dspipeline_types.go:335]: added

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-667`

**Pull request merged**